File: sdk/c/src/ipam.c
#include "hitechcloud/ipam.h"
#include <stdlib.h>


/* GET /api/service/{id}/htcipam/ips - List IP addresses */
char *hc_ipam_list_ips(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/htcipam/ips", NULL);
}

/* GET /api/service/{id}/htcipam/subnets - List subnets */
char *hc_ipam_list_subnets(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/htcipam/subnets", NULL);
}

/* GET /api/service/{id}/htcipam/rdns - List reverse DNS records */
char *hc_ipam_list_rdns(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/htcipam/rdns", NULL);
}

/* POST /api/service/{id}/htcipam/rdns - Set reverse DNS */
char *hc_ipam_set_rdns(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/htcipam/rdns", NULL);
}

File: sdk/r/R/ipam.R
#' HiTechCloudIPAM: IP addresses, subnets, reverse DNS
#' @param http HttpClient object
#' @return Resource object
#' @export
IpamResource <- function(http) {
  env <- new.env(parent = emptyenv())
  env$http <- http
  class(env) <- "IpamResource"
  env
}

#' GET /api/service/{id}/htcipam/ips - List IP addresses
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listIps <- function(self, ...) {
  self$http$get("/api/service/{id}/htcipam/ips", list(...))
}

#' GET /api/service/{id}/htcipam/subnets - List subnets
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listSubnets <- function(self, ...) {
  self$http$get("/api/service/{id}/htcipam/subnets", list(...))
}

#' GET /api/service/{id}/htcipam/rdns - List reverse DNS records
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listRdns <- function(self, ...) {
  self$http$get("/api/service/{id}/htcipam/rdns", list(...))
}

#' POST /api/service/{id}/htcipam/rdns - Set reverse DNS
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
setRdns <- function(self, ...) {
  self$http$post("/api/service/{id}/htcipam/rdns", list(...))
}

